psl/psl/src/lib.rs

```diff
@@ -49,6 +49,14 @@ pub fn parse_configuration_multi_file(
     psl_core::parse_configuration_multi_file(files, builtin_connectors::BUILTIN_CONNECTORS)
 }
 
+/// Parses and validates Prisma schemas, but skip analyzing everything except datasource and generator
+/// blocks. Never fails. In case of errors, tries to return as much of configuration as it was able to parse
+/// and reports the errors in `Diagnostics` params. Consumers then may decided themselves if they want
+/// to use incomplete configuration or convert `Diagnostics` into error.
```

Comment:
Slight nit, mostly because the `in case of errors` looked a bit confusing to me:

```suggestion
/// blocks. It never fails, but when the returned `Diagnostics` contains errors, it implies that the 
/// `Configuration` content is partial.  
/// Consumers may then decide  whether to convert `Diagnostics` into an error.
```